src/Hubbup.MikLabelModel/MikLabelerPredictor.cs

```diff
@@ -10,35 +11,77 @@ namespace Hubbup.MikLabelModel
         {
-            var aspnetIssue = new GitHubIssue
+            var userMentions = _regex.Matches(issue.Body).Select(x => x.Value).ToArray();
+
+            List<LabelAreaScore> labelPredictions;
+            if (filePaths == null)
             {
-                ID = issue.Number.ToString(),
-                Title = issue.Title,
-                Description = issue.Body
-            };
+                var aspnetIssue = new GitHubIssue
+                {
+                    ID = issue.Number,
+                    Title = issue.Title,
+                    Description = issue.Body,
+                    IsPR = 0,
+                    Author = issue.User.Login,
+                    UserMentions = string.Join(' ', userMentions),
+                    NumMentions = userMentions.Length,
+                };
+                var prediction = _predictionEngine.Predict(aspnetIssue);
+                labelPredictions = GetBestThreePredictions(prediction, forPrs: false);
+            }
+            else
+            {
+                var diffHelper = new DiffHelper();
+                var segmentedDiff = diffHelper.SegmentDiff(filePaths);
+                var aspnetIssue = new GitHubPullRequest
+                {
+                    ID = issue.Number,
+                    Title = issue.Title,
+                    Description = issue.Body,
+                    IsPR = 1,
+                    Author = issue.User.Login,
+                    UserMentions = string.Join(' ', userMentions),
+                    NumMentions = userMentions.Length,
+                    FileCount = filePaths.Length,
+                    Files = string.Join(' ', segmentedDiff.FileDiffs),
+                    Filenames = string.Join(' ', segmentedDiff.Filenames),
+                    FileExtensions = string.Join(' ', segmentedDiff.Extensions),
+                    FolderNames = diffHelper.FlattenWithWhitespace(segmentedDiff.FolderNames),
+                    Folders = diffHelper.FlattenWithWhitespace(segmentedDiff.Folders)
+                };
+                var prediction = _prPredictionEngine.Predict(aspnetIssue);
+                labelPredictions = GetBestThreePredictions(prediction, forPrs: true);
+            }
 
-            var prediction = _predictionEngine.Predict(aspnetIssue);
-            var labelPredictions = GetBestThreePredictions(prediction);
             return new LabelSuggestion
             {
                 LabelScores = labelPredictions,
             };
         }
 
-        private List<LabelAreaScore> GetBestThreePredictions(GitHubIssuePrediction prediction)
+        private List<LabelAreaScore> GetBestThreePredictions(GitHubIssuePrediction prediction, bool forPrs)
         {
             var scores = prediction.Score;
 
             VBuffer<ReadOnlyMemory<char>> slotNames = default;
             _predictionEngine.OutputSchema[nameof(GitHubIssuePrediction.Score)].GetSlotNames(ref slotNames);
+            if (forPrs)
```

Comment:
Should this also be an if-else? It seems like the logic is exclusive between issues and PRs.